### PR TITLE
Remove ie8+9 support

### PR DESCRIPTION
**Pull request by Henni**
I looked at all occurences of "ie8" and "ie9" and tried to remove them. There are still some occurences left. But most of the small hacks are removed.

Left to do (probably in a different PR):
- [ ] [message.html](https://github.com/owncloud/core/blob/master/core/templates/message.html#L1) - I'm not completely sure if this only applies to IE8
- [ ] placeholder polyfill
- [ ] window.history polyfill
- [ ] SVG polyfill (+ all PNG files)

### CODE REVIEW

**Comment by DeepDiver1975** [comment]
@pvince81 @dragotin 9.1 topic from my pov

File: core/js/js.js [hunk 1]
```diff
@@ -1892,15 +1860,6 @@ OC.Util = {
 	},
 
 	/**
-	 * Returns whether this is IE8
-	 *
-	 * @return {bool} true if this is IE8, false otherwise
-	 */
-	isIE8: function() {
```
Comment (PVince81) [inline core/js/js.js]: I suggest keeping but deprecating, in case apps decided to use this function.

**Comment by PVince81** [comment]
Code looks good so far.

Note that ViewController also has some IE8 specific code but we might need to keep it in case people still have old IE8-style URLs (with everything in the hash portion)

**Comment by PVince81** [comment]
Needs a bit of regression testing, but from looking at the code it seems the impact should be negligible

File: core/js/js.js (hunk 1 above)
Comment (Henni) [inline core/js/js.js]: @PVince81 deprecation via `console.warn('function isIE8() is deprecated')`?

Comment (PVince81) [inline core/js/js.js]: Yeah, also add a `@deprecated` JSDoc tag.

**Comment by PVince81** [comment]
I don't have time to do regression testing for other IEs.

I know that most safe CSS had an explicit class, so removing these should be safe.
Not sure about the other ones, would be good to test in IE11 still.

Or reduce this PR to only remove the ie8/ie9 classes for now.

**Comment by PVince81** [comment]
I'm moving this to 9.2 for now.

If you have time and do a PR to 9.1 with only the CSS ".ie8" and ".ie9" removed, it will be acceptable for 9.1 before the RC / code freeze.
Thanks.

**Comment by Henni** [comment]
@PVince81 this PR only removed ie8 and ie9 classes. There isn't anything that changes for ie11 and other browsers or did I miss something?

File: core/css/apps.css [hunk 2]
```diff
@@ -162,7 +162,6 @@
 		text-overflow: ellipsis;
 		overflow: hidden;
 		display: inline-block;
-		width: 201px; /* fallback for IE8 */
```
Comment (PVince81) [inline core/css/apps.css]: @Henni how safe is this change ?

Comment (Henni) [inline core/css/apps.css]: supported since IE 9 ([caniuse](http://caniuse.com/#search=calc), [MDN](https://developer.mozilla.org/en/docs/Web/CSS/calc))

**Comment by PVince81** [comment]
Okay, I did a quick test in IE11, personal page, apps page and others, seems to work fine :+1:

**Comment by Henni** [comment]
Reintroduced isIE8 function and deprecated it.
Squashed and rebased.

**Comment by PVince81** [comment]
@Henni thanks :+1:

**Comment by PVince81** [comment]
@owncloud/designers second review please

**Comment by Henni** [comment]
Just found a bug in my own code:
The file upload uses `OC.Util.scaleFixForIE8` which I removed instead of deprecating. I'll fix this in a second.

**Comment by PVince81** [comment]
:+1:

Second review ? @ChristophWurst

**Comment by MorrisJobke** [comment]
Tested and changes look good 👍

**Pull request merged**